ovirt_imageio/client/_ovirt.py

```diff
@@ -357,6 +414,20 @@ def finalize_transfer(con, transfer, disk, timeout=300):
              transfer.id, time.monotonic() - start)
 
 
+def _wait_for_disk(con, disk_id):
+    log.info("Waiting for disk %s", disk_id)
+    timeout = 120
+    start = time.monotonic()
+    deadline = start + timeout
+    disk_service = _disk_service(con, disk_id)
+    while True:
+        time.sleep(1)
```

Comment:
True! Maybe would be even better to have the condition in the while:
```python
    while disk_service.get().status != types.DiskStatus.OK:
        time.sleep(1)
```
That way we avoid the wasted second and some lines of code.

Reply:
Wait, `while` actually always does at least one loop if I remember it correctly :/

So... I will move the sleep to the end and keep the while as is.

Reply:
The disk cannot be ready so fast. Creating a disk takes 8-16 seconds, so sleeping at the start makes sense. Actually it make sense to wait longer before the first check, but if engine fixes the stupid delays, disk can be ready in one second, so this is not the place to fix this.

Some APIs are ready immediately - creating a transfer is actually a blocking call and when it returns the transfer is ready to use.

Reply:
This is a bad pattern:

    while disk_service.get().status != types.DiskStatus.OK:
        ...

Since disk can be deleted while you wait, and we need to handle all possible statuses.

See how we wait for the disk in existing code, this code is well tested (but not complete in some cases, e.g. not checking for ILLEGAL status).